packages/plugin-ext/src/plugin/plugin-manager.ts

```diff
@@ -412,9 +412,10 @@ export class PluginManagerExtImpl implements PluginManagerExt, PluginManager {
 
         if (this.activatedPlugins.get(pluginId)) {
             deferred.resolve();
+            return deferred.promise;
         }
         this.pluginActivationPromises.set(pluginId, deferred);
```

Comment:
is this map set only when calling activate plugin programatically?

Reply:
No, it's being set for all activate plugin methods.